src/gwproxy/gwproxy.c

```diff
@@ -392,18 +393,38 @@ static int parse_options(int argc, char *argv[], struct gwp_cfg *cfg)
 		return -EINVAL;
 	}
 
-	if (cfg->nr_workers <= 0) {
-		fprintf(stderr, "Error: --nr-workers must be at least 1.\n");
+	if (cfg->nr_workers <= 0 || cfg->nr_workers > 1000) {
+		fprintf(stderr, "Error: --nr-workers must be between 1 and 1000.\n");
 		return -EINVAL;
 	}
 
-	if (cfg->target_buf_size <= 1) {
-		fprintf(stderr, "Error: --target-buf-size must be greater than 1.\n");
+	if (cfg->nr_dns_workers <= 0 || cfg->nr_dns_workers > 100) {
+		fprintf(stderr, "Error: --nr-dns-workers must be between 1 and 100.\n");
 		return -EINVAL;
 	}
 
-	if (cfg->client_buf_size <= 1) {
-		fprintf(stderr, "Error: --client-buf-size must be greater than 1.\n");
+	if (cfg->target_buf_size <= 1 || cfg->target_buf_size > (1024 * 1024)) {
+		fprintf(stderr, "Error: --target-buf-size must be between 2 and 1MB.\n");
+		return -EINVAL;
+	}
+
+	if (cfg->client_buf_size <= 1 || cfg->client_buf_size > (1024 * 1024)) {
+		fprintf(stderr, "Error: --client-buf-size must be between 2 and 1MB.\n");
+		return -EINVAL;
+	}
```

Comment:
Allow the buffer size to be set to the maximum size of 5GB.

Also, add a rule: when using SOCKS5, the minimum buffer size must be at least 1024 bytes to accommodate the SOCKS5 handshake, authentication, and command.